datamodel/low/v3/create_document.go

```diff
@@ -45,6 +46,7 @@ func createDocument(info *datamodel.SpecInfo, config *datamodel.DocumentConfigur
 	idxConfig.BasePath = config.BasePath
 	idxConfig.Logger = config.Logger
 	rolodex := index.NewRolodex(idxConfig)
+	<-info.GetJSONParsingChannel() // Need to wait for JSON parsing to complete before we can index.
```

Comment:
I want to double check this does not slow down vacuum too much, I will pull the branch and perform some a/b testing.

Reply:
This is a problem in vacuum for sure. after A/B testing Stripe:

Before change:
`vacuum took 820 milliseconds to lint 4.28 MB across 1 files`

After change: 
`vacuum took 1.12 seconds to lint 4.28 MB across 1 files`

Same with asana:

Before: 
`vacuum took 217 milliseconds to lint 2.07 MB across 1 files`

After :
`vacuum took 329 milliseconds to lint 2.07 MB across 1 files`

The rolodex is not doing any resolving here, so why do we need to wait?

Reply:
so what we were seeing before this change went in was the yaml.Node structure of the document referred to by the RootNode was being modified while still trying to be parsed to JSON within the extract function, which led to infinite loops within that parsing and then the callstack would blow up.

Reply:
This looks like a deeper race condition in the resolver or the rolodex. Lets keep this in place as a safety net, until the root cause can be discovered.